girdermedviewer/app/girder/utils.py

```diff
@@ -42,22 +43,37 @@ def __del__(self):
     def get_item_files(self, item):
         return self.girder_client.listFile(item["_id"])
 
-    @contextmanager
     def download_file(self, file):
-        """
-        :param forced_cache can be used to override the cache mode for a specific file
-        """
-        file_path = os.path.join(self.temp_dir_path, file["_id"], file["name"])
+        file_path = os.path.join(self.temp_dir_path, file['_id'], file["name"])
         if not os.path.exists(file_path):
             logger.info(f"Download {file['name']} to {file_path}")
             self.girder_client.downloadFile(
                 file["_id"],
                 file_path
             )
+        return file_path
+
+    @contextmanager
+    def fetch_file(self, file):
+        fetched_from_assetstore = False
+        clear_cache_required = False
+        if self.assetstore_dir_path:
+            if "path" not in file:
+                logger.info("The file path in assetstore is unknown. Make sure to use the girdermedviewer-plugin")
+            file_path = os.path.join(self.assetstore_dir_path, file['path'])
+            if os.path.exists(file_path):
+                fetched_from_assetstore = True
+            else:
+                logger.info("The file cannot be read from the assetstore, it will be downloaded instead")
+
+        if not fetched_from_assetstore:
+            file_path = self.download_file(file)
+            clear_cache_required = self.cache == CacheMode.No
+
         try:
             yield file_path
         finally:
-            if self.cache == CacheMode.No:
+            if clear_cache_required:

```

Comment:
looking at clearCache, it's fine to call `clear_cache` even if the file is in the assetstore, `clearCache()` won't harm. (maybe we should just enforce that temp_dir_path is != from assetstore_dir_path.
And that `file_path` in `clear_cache` may not exist (because file is in `assetstore_dir_path` instead.